Initium-Odp/src/com/universeprojects/miniup/server/controllers/ViewCharacterMiniController.java

```diff
@@ -0,0 +1,219 @@
+package com.universeprojects.miniup.server.controllers;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Collections;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.cacheddatastore.ShardedCounterService;
+import com.universeprojects.miniup.CommonChecks;
+import com.universeprojects.miniup.server.GameUtils;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.web.Controller;
+import com.universeprojects.web.PageController;
+
+/**
+ * Controller for the view_profile page.
+ * Available request params:
+ *
+**/
+
+@Controller
+public class ViewCharacterMiniController extends PageController {
+
+	public ViewCharacterMiniController() {
+		super("viewcharactermini");
+	}
+	
+	@Override
+	protected String processRequest(HttpServletRequest request,
+			HttpServletResponse response) throws ServletException, IOException {
+		response.setHeader("Access-Control-Allow-Origin", "*");     // This is absolutely necessary for phonegap to work
+		
+		ODPDBAccess db = ODPDBAccess.getInstance(request);
+		CachedEntity character = db.getCurrentCharacter();
+		CachedEntity user = db.getCurrentUser();
+	
+	/**
+	boolean isSelf = false;
+	if ((user!=null && auth.getAuthenticatedUserId()!=null && auth.getAuthenticatedUserId()==user.getKey().getId()) || 
+			(auth.getAuthenticatedInstantCharacterId()!=null && auth.getAuthenticatedInstantCharacterId().equals(characterKey.getId())))
+		isSelf = true;
+	request.setAttribute("isSelf", isSelf); 
+	**/
+	
+	boolean isCloaked = false;
+	if (GameUtils.equals(character.getProperty("cloaked"), true))
+		isCloaked = true;
+	request.setAttribute("isCloaked", isCloaked);
+	
+	
+	String type = (String)character.getProperty("type");
+	if (type==null || type.equals(""))
+		type = "PC";
+	request.setAttribute("type", type);		
+
+	// If the character we're trying to inspect is an NPC and he is still alive, don't allow the inspection
+	if (type.equals("NPC") && (Double)character.getProperty("hitpoints")>0d)
+		throw new RuntimeException("Should not have been able to view this character.");
+	
+	boolean isSelf = GameUtils.equals(db.getCurrentUser().getKey(), character.getProperty("userKey"));
```

Comment:
We've already retrieved the current user at the top of this method (`user` variable). Shortcut to `user.getKey()` instead of `db.getCurrentUser().getKey()`.